ark/schema/shared/jsonSchema.ts

```diff
@@ -34,51 +37,90 @@ export declare namespace JsonSchema {
 		examples?: readonly t[]
 	}
 
-	export type Branch = Constrainable | Const | String | Numeric | Object | Array
+	type Composition = Union | OneOf | Intersection | Not
+
+	type NonBooleanBranch =
```

Comment:
Maybe we can call this something like `CheckedBranch` since `unknown` and `never` don't require looking at the value?

When I see `NonBooleanBranch` I think about excluding the actual data type.

Reply:
> When I see `NonBooleanBranch` I think about excluding the actual data type.

That's exactly what this `NonBooleanBranch` is doing. JSON Schema allows `true` and `false` as valid schemas, where `true` allows any valid json and `false` doesn't allow anything. We have `NonBooleanBranch` for removing those and requiring it to be a "traditional"  schema e.g. `{type: "string"}`

Reply:
I know, but intuitively I think of it as meaning `JsonObject | JsonArray | string | number | null`.